source/loader/layers/sanitizer/msan/msan_interceptor.cpp

```diff
@@ -75,6 +75,7 @@ ur_result_t MsanInterceptor::allocateMemory(ur_context_handle_t Context,
   if (Type != AllocType::DEVICE_USM) {
     return UR_RESULT_SUCCESS;
   }
+  assert(Device);
```

Comment:
I think Device is allowed to be null here since it's a helper that can be used for host allocations as well

Reply:
If that is the case, then Type should be `AllocType::HOST_USM` and should have already returned.